File: by-kotlin/kt-boj/solutions/1929-소수 구하기.kt
import java.util.*
import kotlin.math.sqrt

fun main() = with(Scanner(System.`in`)) {
    val m = nextInt()
    val n = nextInt()

    val result = getPrimes(m, n)
    for (i in m..n) {
        if (result[i]) println(i)
    }
}

fun getPrimes(m: Int, n: Int): BooleanArray {
    val sieve = BooleanArray(n + 1) { true }
    sieve[1] = false

    val k = sqrt(n.toFloat()).toInt()
    for (i in 2..k) {
        if (sieve[i]) {
            for (j in i + i..n step i) {
                sieve[j] = false
            }
        }
    }
    return sieve
}

File: by-python/py-boj/1929-소수 구하기.py
def get_primes(m, n):
    sieve = [True] * (n+1)
    sieve[1] = False 
    k = int(n ** .5)
    
    for i in range(2, k+1):
        if sieve[i]:
            for j in range(i+i, n+1, i):
                sieve[j] = False
    return [w for w in range(m, n+1) if sieve[w]]
 
M, N = map(int, input().split())
S = get_primes(M, N)
 
for s in S:
    print(s)